sigstore-java/src/main/java/dev/sigstore/tuf/model/TufMeta.java

```diff
@@ -15,16 +15,23 @@
  */
 package dev.sigstore.tuf.model;
 
-import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
 import org.immutables.gson.Gson;
 
 /**
  * Generic Tuf Metadata interface for various TUF resources such as Roles, Snapshots, and Targets.
  */
 public interface TufMeta {
+  /** Returns the metadata type. In this case 'root'. */
+  @Gson.Named("_type")
+  String getType();
 
   /** Date at which this data expires. */
-  LocalDateTime getExpires();
+  String getExpires();
+
+  default ZonedDateTime getExpiresAsDate() {
```

Comment:
you can't map two fields to the same json element (I tried). Is that what you were suggestion?

Reply:
A derived value is an immutables construct that calculates this once instead of invoking every time: https://immutables.github.io/immutable.html#derived-attributes

Reply:
hmm. I like the idea! I'm not super psyched about changing this to an abstract class though as it blows up the whole interface hierarchy, we'd have to change all the sub types to abstract as well. WDYT?